src/configuration_proxy.rs

```diff
@@ -149,15 +153,28 @@ impl<T: crate::transport::Transport> ConfigurationProxy<T> {
     async fn handle_message_check_in_now(
         &mut self,
         session_properties: Map,
-        reply: OneshotSender<FeatureFacts>,
+        reply: OneshotSender<(Option<Checkin>, FeatureFacts)>,
     ) -> Result<(), ConfigurationProxyError> {
-        if let Ok(fresh_checkin) = self
+        let fresh_checkin: Option<Checkin> = self
             .transport
             .checkin(session_properties)
             .await
             .inspect_err(|e| tracing::debug!(%e, "Error refreshing checkin configuration"))
-        {
-            self.checkin = Some(fresh_checkin);
+            .ok();
+
+        let changed = fresh_checkin.is_some() && fresh_checkin != self.checkin;
+
+        tracing::trace!(
+            changed,
+            cached = ?self.checkin,
+            fresh = ?fresh_checkin,
+            "Checked in"
+        );
+
+        if changed {
+            if let Some(fresh) = fresh_checkin {
+                self.checkin.replace(fresh);
+            }
```

Comment:
If the fresh checkin is none, would we not want to set that here too?

i.e. self.checkin = fresh_checkin

Reply:
No, because it might be None due to an error like a timeout.